datadog/resource_datadog_role.go

```diff
@@ -15,8 +15,13 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
 )
 
-// validPermissions is a map of all unrestricted permission IDs to their name
-var validPermissions map[string]string
+type PermAttributes struct {
+	Name         string
+	IsRestricted bool
```

Comment:
jw why did we decide to call the attribute `restricted` but the role opt-out field is called `default`? i'd imagine these should be the same word. especially now since the permission is no longer fully restricted

Reply:
yeah this is a good point, i called it `restricted` here because in the `datadog_permissions` resource we also use the term "restricted"

i think we _should_ be using `default` here but i worry about the consistency w/ our other resource(s). if this isn't a major concern tho i'll change it here